transport/tchannel/inbound.go

```diff
@@ -69,7 +69,7 @@ func (i *inbound) Channel() *tchannel.Channel {
 	return i.ch
 }
 
-func (i *inbound) Start(h transport.Handler) error {
+func (i *inbound) Start(h transport.Handler, d transport.Dependencies) error {
```

Comment:
Handler is an Inbound-only dependency. I don't think we should put it on Dependencies. It adds yet another thing that can be nillable in many cases.